util/audio.py

```diff
@@ -1,31 +1,124 @@
 import os
+import io
 import sox
 import wave
+import opuslib
 import tempfile
 import collections
+import numpy as np
+
 from webrtcvad import Vad
+from util.helpers import LimitingPool
 
 DEFAULT_RATE = 16000
 DEFAULT_CHANNELS = 1
 DEFAULT_WIDTH = 2
 DEFAULT_FORMAT = (DEFAULT_RATE, DEFAULT_CHANNELS, DEFAULT_WIDTH)
 
+AUDIO_TYPE_NP = 'application/x-np'
+AUDIO_TYPE_PCM = 'application/x-pcm'
+AUDIO_TYPE_WAV = 'audio/wav'
+AUDIO_TYPE_OPUS = 'application/x-opus'
```

Comment:
FWIW something I just accidentally ran into today: https://tools.ietf.org/html/bcp178

Reply:
The full discussion: https://groups.google.com/forum/#!topic/mozilla.dev.platform/mc-JJFw85HM

Further down the suggest using a vendor prefix: `application/vnd.mozilla.FOO`

Reply:
Changed it now.